sql/analyzer/stored_procedures.go

```diff
@@ -382,8 +382,9 @@ func applyProcedures(ctx *sql.Context, a *Analyzer, n sql.Node, scope *Scope, se
 func applyProceduresCall(ctx *sql.Context, a *Analyzer, call *plan.Call, scope *Scope, sel RuleSelector) (sql.Node, transform.TreeIdentity, error) {
 	pRef := expression.NewProcedureParamReference()
 	call = call.WithParamReference(pRef)
+	currentDatabase := ctx.GetCurrentDatabase()
 
-	esp, err := a.Catalog.ExternalStoredProcedure(ctx, call.Name, len(call.Params))
+	esp, err := a.Catalog.ExternalStoredProcedure(ctx, currentDatabase, call.Name, len(call.Params))
```

Comment:
Gotcha, that makes sense. Let me try out this approach and verify that `dolt_clone` works without issues in the dolt repo.

Reply:
Updating `ShowCreateProcedure::RowIter` is not going to work, since `ShowCreateProcedure::Resolved` returns false (due to no DB) before `RowIter` is ever called.

Reply:
Reworked the PR entirely and force-pushed the new branch. This conversation is no longer relevant to the PR (but it's got great info!).